gradle-baseline-java/src/main/groovy/com/palantir/baseline/extensions/BaselineJavaVersionsExtension.java

```diff
@@ -18,18 +18,20 @@
 
 import javax.inject.Inject;
 import org.gradle.api.Project;
+import org.gradle.api.provider.MapProperty;
 import org.gradle.api.provider.Property;
+import org.gradle.jvm.toolchain.JavaInstallationMetadata;
 import org.gradle.jvm.toolchain.JavaLanguageVersion;
 
 /**
  * Extension named {@code javaVersions} on the root project used to configure all java modules
  * with consistent java toolchains.
  */
 public class BaselineJavaVersionsExtension {
-
     private final Property<JavaLanguageVersion> libraryTarget;
     private final Property<JavaLanguageVersion> distributionTarget;
     private final Property<JavaLanguageVersion> runtime;
+    private final MapProperty<JavaLanguageVersion, JavaInstallationMetadata> jdks;
```

Comment:
I wonder if the map will come back to haunt us? I suspect our jdk repository layouts will be predictable as new language versions are released, we may be able to provide something like a function from `JavaLanguageVersion` -> `Provider<JavaInstallationMetadata>` without requiring custom/internal toolchain plugins to be updated for every release.

Reply:
A map is a function!

But seriously, this is effectively the same as registering a function with this class then calling it to work out the metadata. Pretty sure that's how `MapProperty.put(All)` works under the scenes.